scp/lib/drawer_screens/settings.dart

```diff
@@ -71,6 +71,8 @@ class _SettingsState extends State<Settings> {
   Future fetchUserData(BuildContext context) async {
     SharedPreferences prefs = await SharedPreferences.getInstance();
     phoneNo = prefs.getString('phone_no');
+    username = prefs.getString('username');
+    rollNo = prefs.getString('roll_no');
     DocumentSnapshot qn =
         await Firestore.instance.collection('tokens').document(phoneNo).get();
     state = await qn.data['optionalSub'];
```

Comment:
state is set in fetchUserdata, while state is being checked for null and set true in the initstate. Initstate fires up before fetchUserData function. You sure this is your flow because it looks like it may be prone to error